solidity/contracts/BondedECDSAKeepFactory.sol

```diff
@@ -271,9 +272,38 @@ contract BondedECDSAKeepFactory is IBondedECDSAKeepFactory, CloneFactory {
             );
         }
 
+        keeps.push(address(keep));
+        /* solium-disable-next-line security/no-block-members*/
+        keepOpenedTimestamp[address(keep)] = block.timestamp;
+
         emit BondedECDSAKeepCreated(keepAddress, members, _owner, application);
     }
 
+    /// @notice Gets how many keeps have been opened by this contract.
+    /// @dev    Checks the size of the keeps array.
+    /// @return The number of keeps opened so far.
+    function getKeepCount() external view returns (uint256) {
+        return keeps.length;
+    }
+
+    /// @notice Gets a specific keep address at a given index.
+    /// @return The address of the keep at the given index.
+    function getKeepAtIndex(uint256 index) external view returns (address) {
+        require(index < keeps.length, "Out of bounds.");
+        return keeps[index];
+    }
+
+    /// @notice Gets the opened timestamp of the given keep.
+    /// @return Timestamp the given keep was opened at or 0 if this keep
+    /// was not created by this factory.
+    function getKeepOpenedTimestamp(address _keep)
+        external
+        view
+        returns (uint256)
+    {
+        return keepOpenedTimestamp[_keep];
+    }
```

Comment:
I assume these functions are required by stake drop? Maybe they could also be defined in some interface so there is a reference why they are required in this contract?

Reply:
I'd prefer to not do any decision about it in this PR. I think it's better to postpone it until https://github.com/keep-network/keep-ecdsa/pull/269 gets finalized.